### ISSUE TITLE
Focus is completely lost with Teaching tip

### ISSUE THREAD

**Issue by nilaysanghvi** [microsoft/microsoft-ui-xaml#3257]
I am using the Teaching Tip control in a UWP app. However, I am not able to get focus on the teaching tip. As suggested in the documentation, pressing F6 does not help. The focus is completely lost and I am not able to navigate using the tab key even in the rest of the app. The only option is to press the Esc key which dismisses the teaching tip. 

The reason I suspect this happens is because the "IsLightDismissEnabled" property is set to True, hence there is no cross button on the top right corner. Moreover, I do not use the "ActionButton" or the "CloseButton". If I have any of those, then the focus gets set on pressing F6. I have my own custom content in the Teaching tip - 

```xaml
<muxc:TeachingTip
            x:Name="TestTip"
            Title="Sample Title"
            Target="{x:Bind TaskEntryBar.PlusIcon}"
            PreferredPlacement="TopRight"
            IsLightDismissEnabled="True">
            <Grid>
                <Grid.ColumnDefinitions>
                    <ColumnDefinition Width="*"/>
                    <ColumnDefinition Width="*"/>
                </Grid.ColumnDefinitions>
                <TextBlock Grid.Column="0" Text="2 of 2" FontSize="14"/>
                <Button
                    x:Name="TestButton"
                    Grid.Column="1"
                    FontSize="14"
                    Margin="6,0"
                    Content="Got it"
                    Click="AddTaskTip_Click" />
            </Grid>
</muxc:TeachingTip>
```

![image](https://user-images.githubusercontent.com/70956797/92486647-f6f37800-f209-11ea-88b7-c2a5f1e701ff.png)

As suggested in one of the other [issues](https://github.com/microsoft/microsoft-ui-xaml/issues/2822#issuecomment-655171437), I tried to set the focus ( in the code behind file ) on my custom button but that also did not work - 
```c#
TestTip.IsOpen = true;
TestButton.Focus(FocusState.Programmatic);
```
And using the visual tree helper, I got the popup, but Focus() method is not available on a popup. 
```c#
var popups = VisualTreeHelper.GetOpenPopups(Window.Current);
foreach (var popup in popups)
{
    popup.Focus(FocusState.Programmatic);
}
```

Please suggest what can I do to get focus on the teaching tip in case of only custom content and with IsLightDismissEnabled set to true.

Windows 10 Version: Build 19041.450
Device Form Factor: Desktop

**Comment by StephenLPeters** [comment]
Seems like a few issues. First is an oversight in the F6 logic, that at tip could have no close button, we should fix this so it works with your content.  Second, the isOpen property on teaching tip is async, which means you cannot set it to true and immediately call focus on your content.  You should be able to do this if you attach to the TestButton.Loaded event and call focus in there.  But perhaps we should also add an Opened event to teaching tip, given that IsOpen is async.

**Comment by StephenLPeters** [comment]
Help wanted for the F6 portion.  @SavoySchuler fyi about the opened event.

**Comment by nilaysanghvi** [comment]
@StephenLPeters Thanks, the TestButton.Loaded solution worked. However, the problem with that solution is that the narrator announcement is not made due to the Focus shifting to the button immediately.

**Comment by Felix-Dev** [comment]
I would like to take a look at the Focus keyboard shortcut problem (F6) if that's fine with everybody.

@nilaysanghvi 

>  However, I am not able to get focus on the teaching tip. As suggested in the documentation, pressing F6 does not help

Where exactly did you find that in the documentation I checked the TeachingTip [main documentation](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/dialogs-and-flyouts/teaching-tip) page but found no mention of F6 setting focus into the TeachingTip. I only found such a note [here](https://github.com/microsoft/microsoft-ui-xaml-specs/blob/1fe654b8ea66d99642dc19d8eee99121299dd1fd/active/teachingtip/TeachingTipSpec.md#keyboard-navigation). If this info is not present in the docs, then it should be added.

**Comment by nilaysanghvi** [comment]
@Felix-Dev I'm sorry it is not in the documentation but the [spec](https://github.com/Microsoft/microsoft-ui-xaml-specs/blob/user/savoyschuler/teachingtip/active/teachingtip/TeachingTipSpec.md#keyboard-navigation).
Also if you switch on the narrator service then it announces that you need to press F6.

**Comment by Felix-Dev** [comment]
Definitely should be added to the TeachingTip's main documentation page then. Can open a PR for that later today.

**Comment by nilaysanghvi** [comment]
@Felix-Dev there is another issue regarding the narrator not announcing the subtitle? Should I open a new issue for that?

**Comment by Felix-Dev** [comment]
@nilaysanghvi Yep :)

**Comment by Felix-Dev** [comment]
Since my PR for this issue adds an interaction test I am currently more or less blocked on issue https://github.com/microsoft/microsoft-ui-xaml/issues/3349.

**Comment by joshkoon** [comment]
@StephenLPeters has there been any progress on this? Is this blocked?

**Comment by joshkoon** [comment]
@StephenLPeters is there any traction we can get on this to get it approved/merged?

**Comment by GuildOfCalamity** [comment]
This is especially problematic in a WinUI app for the TextBox controls since is causes the user to have to click the TextBox twice.

The only way I know to get around this is to setup a PropertyChangedCallback for the teaching tip...

`teachingTip.RegisterPropertyChangedCallback(TeachingTip.IsOpenProperty, IsTipOpenChanged);`

The inside the callback shift the focus back to the TextBox.
